### PR TITLE
changefeedccl: default to bare envelope for cdc expressions

### PR DESCRIPTION
**Pull request by HonoreDB**
Changes the default json encoding when using an expression to omit the `after` key and store any metadata in a crdb field. Avro encoding changes "after" to "record".

This is meant as a quick fix to the defaults and validations that inappropriately insist on a metadata-friendly envelope when cdc expressions are meant to provide their own metadata as part of the projection.

Release justification: Change to new functionality.
Release note (bug fix): Improved the default output when using a SELECT clause with a CHANGEFEED.

### CODE REVIEW

**Comment by HonoreDB** [comment]
Splitting this out by request (this shouldn't be blocked by more speculative stuff). Think I'll need to fix a few more tests, and will add some comments in avro.

**Review by miretskiy — CHANGES_REQUESTED** [review]
This looks mostly good; I think addressing testing stuff & adding some commentary in avro encoding
would be good;   Ping once the testing issues cleared out.

> Reviewed 9 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87467#-:-NBOBBvh4HeIVf4g1EFX:bq9ec16)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB and @livlobo)

___
*[`pkg/ccl/changefeedccl/changefeed_test.go` line 2385 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/87467#-NBOB59U60Mo9cSsrbzp:-NBOB59U60Mo9cSsrbzq:bo02vl) ([raw file](https://github.com/cockroachdb/cockroach/blob/78b5a884a2d5ce24225f5788310a2766fab3ed0f/pkg/ccl/changefeedccl/changefeed_test.go#L2385)):*
> ```Go
> 	// TODO: Fix testfeed issues.
> 	//cdcTest(t, testFn, feedTestForceSink("webhook"))
> 	//cdcTest(t, testFn, feedTestForceSink("cloudstorage"))
> ```

What kind of issues?  
___
*[`pkg/ccl/changefeedccl/cdceval/validation.go` line 51 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/87467#-NBNaXKv912U_KmcL3wc:-NBNaXKv912U_KmcL3wd:b-g3z8cx) ([raw file](https://github.com/cockroachdb/cockroach/blob/78b5a884a2d5ce24225f5788310a2766fab3ed0f/pkg/ccl/changefeedccl/cdceval/validation.go#L51)):*
> ```Go
> 			`filters and projections not supported until upgrade to version %s or higher is finalized`,
> 			clusterversion.EnablePredicateProjectionChangefeed.String())
> 	} */
> ```

stray comments?

**Comment by HonoreDB** [comment]
Ready for a look!

**Review by miretskiy — CHANGES_REQUESTED** [review]
> Reviewed 4 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87467)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB, @livlobo, and @miretskiy)

___
*[`pkg/ccl/changefeedccl/encoder_json.go` line 49 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87467#-NBtNASM7-PaQeJMVXXG:-NBtNASM7-PaQeJMVXXH:b-gebc1h) ([raw file](https://github.com/cockroachdb/cockroach/blob/75c231ef1565affbc9594526aa36bd4bcbb02ea1/pkg/ccl/changefeedccl/encoder_json.go#L49)):*
> ```Go
> func canJSONEncodeMetadata(e changefeedbase.EnvelopeType) bool {
> 	// bare envelopes use the _crdb_ key, wrapped envelopes put metadata keys at the top level
> 	// because columns aren't.
> ```

can you rephrase the comment a bit?

**Review by miretskiy — APPROVED** [review]
Looks Good!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87467#-:-NBtNOu26HMNqrBWr3aP:b-w9n4r1)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB, @livlobo, and @miretskiy)

**Comment by HonoreDB** [comment]
Comments addressed, thanks for the review!

**Comment by HonoreDB** [comment]
bors r=[miretskiy]
